client/allocrunner/taskrunner/template/template.go

```diff
@@ -132,6 +133,8 @@ type TaskTemplateManagerConfig struct {
 	TaskID string
 
 	Logger hclog.Logger
+
+	RenderFunc renderer.Renderer
```

Comment:
Could we add a comment here explaining why we have this custom func? It would be useful to add the context from the work you've done into this change.